nncf/experimental/openvino_native/graph/model_transformer.py

```diff
@@ -0,0 +1,111 @@
+        """
+        output_insert_transformations = []
+        transformations = transformation_layout.transformations
+
+        for transformation in transformations:
+            if isinstance(transformation, OVOutputInsertionCommand):
+                output_insert_transformations.append(transformation)
+
+        if output_insert_transformations:
+            self._apply_output_insertion_transformations(output_insert_transformations)
+
+        return self._model
+
+    def _apply_output_insertion_transformations(self, transformations: List[OVOutputInsertionCommand]) -> None:
+        """
+        Applies incoming transformations to the model.
+
+        :param transformations: list of the OVOutputInsertionCommand transformations.
+        """
+        extra_model_outputs = self._get_extra_model_outputs(transformations)
+        model_with_intermediate_outputs = self._insert_outputs(self._model, outputs=extra_model_outputs)
+        self._model = model_with_intermediate_outputs
+
+    def _get_extra_model_outputs(self,
+                                 transformations: List[OVOutputInsertionCommand]) -> List[ov.Output]:
+        """
+        Collects extra model outputs based on transformations.
+
+        :param transformations: lisf of the OVOutputInsertionCommand.
+        :return: list of the output names.
+        """
+
+        extra_model_outputs = []
+        for transformation in transformations:
+            node_name = transformation.target_point.target_node_name
+            node = self.name_to_node_mapping[node_name]
+            port_id = transformation.target_point.port_id
+            if transformation.target_point.type == TargetType.POST_LAYER_OPERATION:
+                output = node.output(port_id)
+                extra_model_outputs.append(output)
+            elif transformation.target_point.type == TargetType.PRE_LAYER_OPERATION:
+                output = node.input_value(port_id)
+                extra_model_outputs.append(output)
+            else:
+                raise NotImplementedError(f'Unsupported target point type {transformation.target_point.type}')
+
+        return extra_model_outputs
+
+    def _insert_outputs(self, model: ov.Model, outputs: List[ov.Output] = None) -> ov.Model:
+        """
+        Takes a model and adds outputs based on the list of ov.Output.
+
+        :param model: OpenVINO model.
+        :param outputs: list of ov.Output.
+        :return: modified model.
+        """
+        if outputs is None:
+            raise RuntimeError("Parameter outputs cannot be None.")
+
+        model_outputs = model.get_results()
+        params = model.get_parameters()
+        extra_model_outputs = []
+        for output in outputs:
+            output_name = output.get_node().get_friendly_name()
+            port_id = output.get_index()
+            result = ov.opset9.result(output, name=f'Result_{output_name}.{port_id}')
```

Comment:
Does using `opset9` here mean we'll have to update it with the next new set of operations?

Reply:
I think so. This implementation was recommended in ticket: 91084.

Reply:
I see. This comment is not related to the PR, I suppose. The current solution doesn't seem valid to me. It would be better to have something like `latest_opset` or etc.
@AlexKoff88, what do you think?

Reply:
the recommendation is to use the latest opset but do it in the following way:
```
from openvino.runtime import opset9 as opset
...
result = opset.result(...)
```

Reply:
According to 
https://github.com/openvinotoolkit/openvino/blob/c09af40b4c089595b57c64d9e7271c63149be9de/src/bindings/python/src/openvino/runtime/__init__.py#L58 `opset10` is available. I switched to it.

Reply:
Reverted to `opset9` due to CI using `openvino-dev` from pypi.